src/utils/math.js

```diff
@@ -95,3 +95,37 @@ export function mod(a, b) {
 export function clamp(val, min, max) {
   return Math.min(Math.max(val, min), max);
 }
+
+/**
+ * Returns value bound to min and max values +/- extent.
+ * @param {number} val the value to bound.
+ * @param {number} min the lower bound.
+ * @param {number} max the upper bound
+ * @param {number} extent the allowed extent beyond the bounds.
+ * @return {number} the bounded value.
+ */
+export function boundValue(val, min, max, extent) {
+  return clamp(val, min - extent, max + extent);
```

Comment:
Probably worth adding a dev assert that min <= max.

Reply:
If that's the case, it seems to make the most sense to put that in `clamp` rather than here

Reply:
Added to both